pkg/exporter/sumologicexporter/exporter.go

```diff
@@ -282,7 +282,9 @@ func (se *sumologicexporter) pushLogsData(ctx context.Context, ld pdata.Logs) er
 				}
 
 				// If metadata differs from currently buffered, flush the buffer
-				if currentMetadata.string() != previousMetadata.string() && previousMetadata.string() != "" {
+				currentMetadataString := fmt.Sprintf("%v", currentMetadata.orig.AsRaw())
+				previousMetadataString := fmt.Sprintf("%v", previousMetadata.orig.AsRaw())
```

Comment:
in this form, please add metadata function for that

Reply:
I actually found another way to compare the objects, using `reflect.DeepEqual()`.

Reply:
still, could we extract it to function? :sweat_smile: